catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java

```diff
@@ -414,12 +405,10 @@ private static void parseTopLevel(RegistryObjectType registryObject, MetacardImp
 
         if (registryObject.isSetExternalIdentifier()) {
             List<ExternalIdentifierType> extIds = registryObject.getExternalIdentifier();
-            for (ExternalIdentifierType extId : extIds) {
-                if (extId.getId()
-                        .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID)) {
-                    metacard.setId(extId.getValue());
-                }
-            }
+            extIds.stream()
+                    .filter(extId -> extId.getId()
+                            .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID))
+                    .forEach(extId -> metacard.setId(extId.getValue()));
```

Comment:
how many extIds do you expect to match? It looks like it's just one (or if multiple match, you overwrite the metacard ID multiple times) I think you could use a stream().findFirst() call here